### DISCUSSION TITLE
Accessing value by native code but getting error

### DISCUSSION THREAD

**Discussion by virtuoso1234** [react-native-async-storage/async-storage#833] [Q&A]
I've followed this docs: https://react-native-async-storage.github.io/async-storage/docs/advanced/brownfield/  for accessing my appTheme in local storage, but in resumeWith(@NotNull Object o), I logged it in logcat, it showed: `Failure(kotlinx.coroutines.JobCancellationException: StandaloneCoroutine has completed normally; job=StandaloneCoroutine{Completed}@7136858)`

I tried both store appTheme by string and stringify but still have error, what did I do wrong?

Here is MyMainActivity's calling: 
![image](https://user-images.githubusercontent.com/106294612/184578238-12f3d6e4-c641-43ab-97fb-01dc9014865f.png)

Reading value function:
![image](https://user-images.githubusercontent.com/106294612/184578283-0ddfc71e-98b0-4380-a133-0c2cbe3e0e5d.png)

**Comment by krizzu** [comment]
The Java example is not recommended and was written there because of requests. I suggest to move to Kotlin, so that it's easier to know what's wrong.

**Comment by virtuoso1234** [comment]
@krizzu I tried but I'm very newbie of kotlin, I searched sth like: "use suspend kotlin function from java" but still didn't find anything

**Comment by tido64** [comment]
@dattmahiho: Does the docs help: https://react-native-async-storage.github.io/async-storage/docs/advanced/brownfield#kotlin-recommended

@krizzu: Is the docs here clear enough? Do we need to add more details?

**Comment by virtuoso1234** [comment]
@tido64 the docs is good but I'm not :( I don't know how to integrate kotlin in java

**Comment by tido64** [comment]
There are some docs here for adding Kotlin to existing projects with samples you can look at: https://developer.android.com/kotlin/add-kotlin

Unfortunately, I'm not much of an Android expert myself so I will convert this issue to a discussion thread so others may be able to help you.

**Comment by virtuoso1234** [comment]
@tido64 I did it, adding kotlin dependencies carefully and call suspend function from java is the key
thank guys

**Reply by maikonalexandre** [reply]
Olá, Tudo bem. 
Estou passando dificuldades com o mesmo problema. Você poderia me dar mais detalhes de como resolveu? 
Desde já, agradeço.

**Comment by virtuoso1234** [comment]
@tido64 sr, but how about ios? I got a exception `Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Must be executed on storage thread'` with these code: 
![image](https://user-images.githubusercontent.com/106294612/188348180-65223744-8dee-4503-8362-bad520478e88.png)

**Reply by tido64** [reply]
You need to use the correct queue, which can be retrived by calling `[asyncStorage methodQueue]`:
```objc
dispatch_async([asyncStorage methodQueue], ^{
    ...
});

**Reply by virtuoso1234** [reply]
I added it then, but *result just returned "("
